### PR TITLE
Don't provide pr_number if it's not a PR

### PR DESCRIPTION
**Pull request by austin3dickey**
We shouldn't be providing `pr_number` to Conbench if the benchmark isn't running on a PR.

### CODE REVIEW

**Review by alistaire47 — APPROVED** [review]
This looks fine. Do we need to mirror this in the other instances of `github_info()`? I think this one is arrow-specific, so I'm not sure we need to change the conbench and benchadapt ones. 

Also, do we know why PR number is set still on default branch builds? Are they still associated with the PR merged?

**Comment by austin3dickey** [comment]
>Do we need to mirror this in the other instances of github_info()?

No, this is specific to how `arrow-benchmarks-ci` provides environment variables to its benchmark builds.

>do we know why PR number is set still on default branch builds?

Let me rephrase: why does `arrow-benchmarks-ci` always pass `BENCHMARKABLE_PR_NUMBER` to its builds, even if it's not a PR? That... is totally my mistake from when I implemented it. You're right, we should fix this on the `arrow-benchmarks-ci` side. Thanks for the helpful question!
